client/lib/posts/post-list-cache-store.js

```diff
@@ -86,6 +87,21 @@ function markDirty( post, oldStatus ) {
 
 		entry.dirty = true;
 	}
+
+	// clear api cache for records with matching site/status
+	cacheIndex.clearRecordsByParamFilter( ( reqParams ) => {
+		const siteIdentifiers = affectedSites.slice( 0, -1 ); // remove the `false` value from above
```

Comment:
As we discussed in slack, this one threw me for a loop a bit.  Guessing it is out of scope for this PR, but a refactor of using `false` in an array to trigger/skip clearing all sites cache might be nice :+1: